knotx-core/src/main/java/io/knotx/dataobjects/KnotContext.java

```diff
@@ -91,6 +94,15 @@ public KnotContext setFragments(List<Fragment> fragments) {
     return this;
   }
 
+  public List<FileData> getFilesData() {
```

Comment:
I'm not familiar with the way this works with Vert.x data objects but we're binding an API to a concrete class here. Perhaps it would be more flexible to use an interface to define the contract of the generic param instead? Or am I overthinking this?